### PR TITLE
Feature/#646 - Support sqlalchemy 2.0 on repository v2

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-core/issues/646

Most of the changes are from https://github.com/Avaiga/taipy-core/pull/647.
However, there are some new changes in the base class of repository v2. I have left the link to reference in the PR.

### CODE REVIEW

File: src/taipy/core/_repository/_sql_model.py [hunk 1]
```diff
@@ -15,7 +15,13 @@
 from sqlalchemy import Boolean, Column, Integer, Text
 from sqlalchemy.ext.declarative import declarative_base
 
-Base = declarative_base()
+
+class Base:
+    __allow_unmapped__ = True
+
+
+Base = declarative_base(cls=Base)  # type: ignore
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_sql_model.py:23]: Reference: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-six-add-allow-unmapped-to-explicitly-typed-orm-models

File: tests/core/repository/mocks.py [hunk 2]
```diff
@@ -24,13 +24,18 @@
 from src.taipy.core._version._version_manager import _VersionManager
 from taipy.config.config import Config
 
-Base = declarative_base()
+
+class Base:
+    __allow_unmapped__ = True
+
+
+Base = declarative_base(cls=Base)  # type: ignore
```
Comment (trgiangdo) [inline tests/core/repository/mocks.py:32]: Reference: https://docs.sqlalchemy.org/en/20/changelog/migration_20.html#migration-to-2-0-step-six-add-allow-unmapped-to-explicitly-typed-orm-models

File: src/taipy/core/_repository/_sql_model.py [hunk 3]
```diff
@@ -29,7 +35,7 @@ class _TaipyModel(Base):  # type: ignore
     model_name = Column(Text)
     document = Column(Text)
 
-    def __int__(self, model_id: str, model_name: str, document: Json):
+    def __init__(self, model_id: str, model_name: str, document: Json):
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_sql_model.py:38]: I guess this is a typo, so I fix it. But I'm not sure.

**Review by toan-quach — APPROVED** [review]
lgtm

**Pull request merged**